File: 백준/Silver/2217. 로프/로프.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
  int N;
  cin >> N;
  int arr[N];
  for (int i = 0; i < N; i++) {
    cin >> arr[i];
  }
  sort(arr, arr+N, greater<>());
  int weight = -1;
  for (int i = 0; i < N; i++) {
    weight = max(weight, arr[i] * (i+1));
  }
  cout << weight;
  return 0;
}

File: 백준/Silver/2217. 로프/로프.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int num = Integer.parseInt(br.readLine());
        int[] arr = new int[num];
        int max = Integer.MIN_VALUE;
        for (int i = 0; i < num; i++) {
            arr[i] = Integer.parseInt(br.readLine());
        }
        Arrays.sort(arr);

        int tmp = 0;
        for (int i = 0; i < arr.length; i++) {
            tmp = arr[i] * (arr.length - i);
            if (max < tmp) {
                max = tmp;
            }
        }
        System.out.println(max);
        /**
         * 오름차순 정렬
         * arr[i] * arr.length - i;
         * 이 값들중 최댓값을 갱신해서 반환
         */
    }
}
